backends/s3/s3.go

```diff
@@ -109,36 +109,29 @@ func (b *Backend) List(ctx context.Context, prefix string) (simpleblob.BlobList,
 	}
 
 	// Request and cache full list, and use marker file to invalidate the cache
-	now := time.Now()
 	data, err := b.Load(ctx, UpdateMarkerFilename)
-	if err != nil && !os.IsNotExist(err) {
-		return nil, err
+	exists := true
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+		exists = false
 	}
 	current := string(data)
 
-	b.mu.Lock()
-	lastMarker := b.lastMarker
-	age := now.Sub(b.lastTime)
-	b.mu.Unlock()
-
-	var blobs simpleblob.BlobList
-	if current != lastMarker || age >= b.opt.UpdateMarkerForceListInterval {
+	if !exists || b.lastMarker == "" || current != b.lastMarker || time.Since(b.lastTime) >= b.opt.UpdateMarkerForceListInterval {
```

Comment:
Any access to `b.last*` requires holding the lock, which is one reason the code created new variables in a critical section.

The other reason is readability.